lib/msf/core/auxiliary/scanner.rb

```diff
@@ -240,19 +240,19 @@ def run
       break if has_fatal_errors?
 
       # Exit if there are no more pending threads
-      if (@tl.length == 0)
+      if (@thread_list.length == 0)
         break
       end
 
       # Attempt to wait for the oldest thread for a second,
       # remove any finished threads from the list
       # and continue on.
       tla = 0
-      @tl.map {|t| tla += t[:batch_size] }
-      @tl.first.join(1)
-      @tl.delete_if { |t| not t.alive? }
+      @thread_list.map {|t| tla += t[:batch_size] if t[:batch_size] }
```

Comment:
If we land this PR, are we okay with the potential of skipping over hosts unintentionally?

i.e. Before this change, a crash was observed - and it was clear that something had gone wrong, and the scan needed to be restarted to verify all of the targets.

After this change, we handle this unknown scenario more gracefully, but potentially there was still issues that stopped hosts from being scanned, and they're not longer surfaced to the user

Reply:
This should not impact the execution, this guard will only impact the calculations to skip `counting` objects in a thread that is not reporting a `:batch_size`.

Execution is based on `@thread_list` containing any threads that are still `.alive?`.

Reply:
I was less worried about the overall host count tracking - and was more worried about why a thread went awol and dropped local storage values, as there's the potential that it's just skipped hosts.

Happy for this to land in the interim though